src/main/java/com/google/sps/server/ScheduleCalendar.java

```diff
@@ -0,0 +1,250 @@
+   *
+   * @param timeInMin the time, in minutes of the week, for an event (either the start or end time)
+   * @param month the current month
+   * @param year the current year
+   * @param nextSun the day of the month of the next Sunday
+   * 
+   * @return a String formatted for DateTime, to be used in creating a calendar event 
+   */
+  public static String calculateDateTime(int timeInMin, int month, int year, int nextSun) {
+    int dayOfWeek = (int) Math.floorDiv(timeInMin, 24 * 60);
+    int dayInHours = dayOfWeek * 24;
+    int hour = (int) Math.floor((timeInMin / 60) - dayInHours);
+    int minute = (int) (60 * (((timeInMin / 60.0) - dayInHours) - hour));
+    int day = dayOfWeek + nextSun;
+     
+    String dayString = day > 9 ? Integer.toString(day) : "0" + Integer.toString(day);
+    String monthString = month > 9 ? Integer.toString(month) : "0" + Integer.toString(month);
+    String hourString = hour > 9 ? Integer.toString(hour) : "0" + Integer.toString(hour);     
+    String minuteString = minute > 9 ? Integer.toString(minute) : "0" + Integer.toString(minute);
+
+    String dateTime = year + "-" + monthString + "-" + dayString + "T" + hourString + ":" + minuteString + ":00";
+    return dateTime;
+  }
+
+   /**
+   * Helper function to calculate the properly formatted start time of a course
+   *
+   * @param timerange the time range for a course's meeting time (ex. 10:30-12:00)
+   * @param month the current month
+   * @param year the current year
+   * @param nextSun the day of the month of the next Sunday
+   * 
+   * @return a string representing the start time
+   */
+  public static String calculateStartTime(TimeRange timerange, int month, int year, int nextSun) {
+    // user will input their semester / quarter start date
+    int startTimeInMin = timerange.start();
+    String startTime = calculateDateTime(startTimeInMin, month, year, nextSun);
+    return startTime;
+  }
+
+   /**
+   * Helper function to calculate the properly formatted end time of a course
+   *
+   * @param timerange the time range for a course's meeting time (ex. 10:30-12:00)
+   * @param month the current month
+   * @param year the current year
+   * @param nextSun the day of the month of the next Sunday
+   * 
+   * @return a string representing the end time
+   */
+  public static String calculateEndTime(TimeRange timerange, int month, int year, int nextSun) {
+    int endTimeInMin = timerange.end();
+    String endTime = calculateDateTime(endTimeInMin, month, year, nextSun);
+    return endTime;
+  }
+
+   /**
+   * Helper function to add a course in the schedule to the calendar, as an event. 
+   *
+   * @param course the course being added 
+   * @param calendarId the id of the secondary calendar to add the course to
+   * @param startTime the start time of the course
+   * @param endTime the end time of the course
+   */
+
+  public void addEvent(Course course, String startTime, String endTime) throws IOException {
```

Comment:
It looks like this is only being used internally within this class, so this should be a private method.